src/error.rs

```diff
@@ -1,20 +1,55 @@
-use std::num::ParseIntError;
-
 use thiserror::Error;
 
+/// Errors that can occur during parsing of
+/// `.libgdx.atlas` files, or during loading of
+/// [`LibGdxAtlasAsset`](crate::LibGdxAtlasAsset)s.
+// The errors have an inner `Box` because
+// the size of the error type is
+// as big as the largest variant,
+// so when the error is bubbled up
+// in the call stack it can allocate
+// more memory than is needed on the
+// stack, if say the error is
+// actually a smaller variant.
 #[non_exhaustive]
 #[derive(Debug, Error)]
 pub enum LibGdxAtlasAssetError {
-    /// An [IO](std::io) Error
+    /// An [IO](std::io) Error that occured
+    /// during parsing of a `.libgdx.atlas` file.
     #[error("Could not load asset: {0}")]
-    Io(#[from] std::io::Error),
+    Io(Box<std::io::Error>),
 
+    /// A Bevy [`LoadDirectError`](bevy::asset::LoadDirectError) that occured
+    /// while loading a [`LibGdxAtlasAsset::image`](crate::LibGdxAtlasAsset::image).
     #[error("Could not load asset: {0}")]
-    LoadDirect(#[from] bevy::asset::LoadDirectError),
+    LoadDirect(Box<bevy::asset::LoadDirectError>),
 
+    /// An error that occurs when parsing the
+    /// content of a `.libgdx.atlas` file.
     #[error("Parse error: {0}")]
     ParsingError(String),
 
+    /// An error that can occur when
+    /// parsing the size of a `.libgdx.atlas`'s
+    /// texture atlas.
     #[error("Parse Int error: {0}")]
-    ParsingInt(#[from] ParseIntError),
+    ParsingInt(Box<std::num::ParseIntError>),
```

Comment:
this one is small, does not need to be boxed.

also we can use the `#[from]` macro here instead of a manual `From` impl, same for `Io`